doc/deps.md

```diff
@@ -0,0 +1,179 @@
+export PATH=<thrift_install_dir>/bin:$PATH
+```
+
+## Known issues
+- In Ubuntu 16.04 LTS the LLVM/Clang has some packaging issues, i.e. some libs
+  are searched in `/usr/lib` however the package has these in
+  `/usr/lib/llvm-3.8/lib` (see http://stackoverflow.com/questions/38171543/error-when-using-cmake-with-llvm).
+  This problem causes an error when emitting `cmake` command during CodeCompass
+  build. A solution would be to download a prebuilt package from the LLVM/Clang
+  webpage but another issue is that the prebuilt packages don't use RTTI which
+  is needed for CodeCompass compilation. So Clang needs to be compiled with
+  RTTI manually:
+
+```bash
+wget http://releases.llvm.org/3.8.0/llvm-3.8.0.src.tar.xz
+tar -xvf llvm-3.8.0.src.tar.xz
+cd llvm-3.8.0.src/tools
+wget http://releases.llvm.org/3.8.0/cfe-3.8.0.src.tar.xz
+tar -xvf cfe-3.8.0.src.tar.xz
+mv cfe-3.8.0.src clang
+rm cfe-3.8.0.src.tar.xz
+cd ../..
+
+mkdir build
+cd build
+export REQUIRES_RTTI=1
+cmake -G "Unix Makefiles" -DLLVM_ENABLE_RTTI=ON ../llvm-3.8.0.src -DCMAKE_INSTALL_PREFIX=<clang_install_dir>
+# This make step takes a while. If you have more CPUs then you can compile on
+# several threads with -j<number_of_threads> flag.
+make install
+
+export CMAKE_PREFIX_PATH=<clang_install_dir>/share/llvm/cmake:$CMAKE_PREFIX_PATH
+```
+
+- As of `gcc` version 5 the ABI has changed which practically means that some
+  symbols in `std` namespace (like `std::string` and `std::list`) contain
+  `__cxx11` in their mangled names. This results linkage errors if the compiled
+  project has libraries compiled with an earlier version of `gcc`.
+
+  In the official Ubuntu 16.04 LTS package repository ODB is stored with the
+  earlier ABI, not like other dependencies such as Boost. The solution is to
+  download and recompile ODB using a new C++ compiler and extend
+  `CMAKE_PREFIX_PATH` environment variable so CMake can find it:
+
+```
+wget http://www.codesynthesis.com/download/odb/2.4/libodb-2.4.0.tar.gz
+tar -xvf libodb-2.4.0.tar.gz
+cd libodb-2.4.0
+./configure --prefix=<odb_install_dir>
+make install
+cd ..
+
+# If you use SQLite
+wget http://www.codesynthesis.com/download/odb/2.4/libodb-sqlite-2.4.0.tar.gz
+tar -xvf libodb-sqlite-2.4.0.tar.gz
+cd libodb-sqlite-2.4.0
+./configure --prefix=<odb_install_dir>
+make install
+cd ..
+
+# If you use PostgreSQL
+sudo apt-get install postgresql-server-dev-<version> # Needed for this step
+wget http://www.codesynthesis.com/download/odb/2.4/libodb-pgsql-2.4.0.tar.gz
+tar -xvf libodb-pgsql-2.4.0.tar.gz
+cd libodb-pgsql-2.4.0
+./configure --prefix=<odb_install_dir>
+make install
+cd ..
+
+sudo apt-get install gcc-<version>-plugin-dev
+sudo apt-get install libcutl-dev
+wget http://www.codesynthesis.com/download/odb/2.4/odb-2.4.0.tar.gz
+tar -xvf odb-2.4.0.tar.gz
+cd odb-2.4.0.tar.gz
+./configure --prefix=<odb_install_dir>
+make install
+cd ..
+
+export CMAKE_PREFIX_PATH=<odb_install_dir>:$CMAKE_PREFIX_PATH
+```
+
+# Build CodeCompass
+
```

Comment:
We have to export the installation lib directory of thrift, odb, and clang for running CodeCompass:
```bash
export LD_LIBRARY_PATH=<thrift_install_dir>/lib:<clang_install_dir>/lib:<odb_install_dir>/lib:$CMAKE_PREFIX_PATH
```